include/gc/Analysis/MatmulConfigAnalysis.h

```diff
@@ -0,0 +1,168 @@
+//===-- MatmulConfigAnalysis.h - the analysis for matmul config -*- C++ -*-===//
+//
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef MLIR_ANALYSIS_MATMULCONFIGANALYSIS_H
+#define MLIR_ANALYSIS_MATMULCONFIGANALYSIS_H
+
+#include "gc/Dialect/Linalgx/LinalgxOps.h"
+#include "mlir/Dialect/DLTI/DLTI.h"
+#include "mlir/Dialect/Linalg/IR/Linalg.h"
+#include "mlir/Interfaces/DataLayoutInterfaces.h"
+
+namespace mlir {
+namespace gc {
+
+using namespace mlir;
+
+struct SystemDesc {
+  // get runtime OMP_NUM_THREADS
+  uint32_t getNumThreads() {
```

Comment:
Could we move this struct to another common header for better reuse?

Reply:
Let's do this when implementing centerized DLTI. You can duplicate this to avoid dependency.

Reply:
Yes, I will do it with another PR for this issue https://github.com/intel/graph-compiler/issues/188 and add more verifer/default value.